skopt/callbacks.py

```diff
@@ -220,3 +220,37 @@ def _criterion(self, result):
 
         else:
             return None
+
+
+class DeadlineStopper(EarlyStopper):
+    """
+    Stop the optimization before running out of a fixed budget of time.
+
+    Attributes
+    ----------
+    * `iter_time`: [list, shape=(n_iter,)]:
+        `iter_time[i-1]` gives the time taken to complete iteration `i`
+
+    Parameters
+    ----------
+    * `total_time`: fixed budget of time (seconds) that the optimization must
+        finish within.
+    """
+    def __init__(self, total_time):
+        super(DeadlineStopper, self).__init__()
+        self._time = time()
+        self.iter_time = []
+        self.total_time = total_time
+
+    def __call__(self, result):
+        elapsed_time = time() - self._time
```

Comment:
You could move all this logic to `_cirterion` if you wanted to and then not have a `__call__`.